File: nhglib-core/src/io/github/voidzombie/nhglib/input/StickType.java
package io.github.voidzombie.nhglib.input;

/**
 * Created by Fausto Napoli on 23/01/2017.
 */
public enum StickType {
    LEFT,
    RIGHT,
    VIRTUAL;

    public static StickType fromString(String value) {
        StickType res = null;

        switch (value.toLowerCase()) {
            case "left":
                res = LEFT;
                break;

            case "right":
                res = RIGHT;
                break;

            case "virtual":
                res = VIRTUAL;
                break;
        }

        return res;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/input/enums/StickType.kt
package io.github.movementspeed.nhglib.input.enums

/**
 * Created by Fausto Napoli on 23/01/2017.
 */
enum class StickType {
    LEFT,
    RIGHT,
    VIRTUAL;

    companion object {
        fun fromString(value: String): StickType? {
            var res: StickType? = null

            when (value.toLowerCase()) {
                "left" -> res = LEFT
                "right" -> res = RIGHT
                "virtual" -> res = VIRTUAL
            }

            return res
        }
    }
}
